### PR TITLE
Fix signal handler to have stable shutdown

### PR DESCRIPTION
**Pull request by coolo**
Do not call stop_vm through JSON but just kill the backend process
and leave the shutdown to itself

### CODE REVIEW

File: isotovideo [hunk 1]
```diff
@@ -73,19 +74,24 @@ bmwqemu::clean_control_files();
...
+my $backend;
 
 # all so ugly ...
 sub signalhandler {
 
-    # do not start a race about the results between the threads
-
     my $sig = shift;
-    diag("signalhandler $$: got $sig");
+    print STDERR "$$: signalhandler got $sig\n";
```
Comment (okurz) [inline isotovideo]: why not diag? maybe duplicate these messages to both log and system journal?

File: isotovideo [hunk 2]
```diff
@@ -73,19 +74,24 @@ bmwqemu::clean_control_files();
...
-        $autotest::running->fail_if_running();
-        $autotest::running = undef;
+	$autotest::running->fail_if_running();
+	$autotest::running = undef;
+    }
+    if (defined $backend && $backend->{backend_pid}) {
+	print STDERR "killing backend process\n";
+        kill('SIGTERM', $backend->{backend_pid});
+        waitpid($backend->{backend_pid}, 0);
+        print STDERR "done with backend process\n";
```
Comment (okurz) [inline isotovideo]: and tidy does not complain about broken indention?

File: isotovideo [hunk 3]
```diff
@@ -95,14 +101,32 @@ sub signalhandler {
...
+        print STDERR "done joining commands process\n";
+	$cpid = 0;
     }
-    exit(1);
+    _exit(1);
 }
 
+sub signalhandler_chld {
+
+    my $sig = shift;
```
Comment (okurz) [inline isotovideo]: `my ($sig) = @_;`?

File: isotovideo (hunk 2 above)
Comment (coolo) [inline isotovideo]: possibly it's not running on isotovideo?

File: isotovideo (hunk 1 above)
Comment (coolo) [inline isotovideo]: this was just my debugging - diag is fine. And no, diag can't write in the system journal - the worker catches all output of isotovideo into autoinst-log.txt

**Comment by coolo** [comment]
Added tidy and diag

File: isotovideo (hunk 1 above)
Comment (okurz) [inline isotovideo]: that's what I am saying, diag is writing to autoinst-log.txt but "print STDERR" might print to syslog so could do
```perl
sub super_important_log {
    my (@args) = @_;
    diag(@args);
    print STDERR "@args\n";
}
…
super_important_log("signalhandler $$: got $sig");
```

**Comment by okurz** [comment]
As I could just observe my s390x worker to fool me, it can't reconnect, i investigated and found a stale worker process to be the culprit. So I will try now your change. Additionally, what do you think about:
* If SIGINT is called a second time or more, try harder to kill ourselves and children
* Spawn an internal "watchdog" on termination and kill children and parent process after a grace period if not shut down already, i.e. KILL after 30s

I did not observe problems when ending the process with SIGQUIT

**Comment by okurz** [comment]
As I could just observe my s390x worker to fool me, it can't reconnect, i investigated and found a stale worker process to be the culprit. So I will try now your change. Additionally, what do you think about:
* If SIGINT is called a second time or more, try harder to kill ourselves and children
* Spawn an internal "watchdog" on termination and kill children and parent process after a grace period if not shut down already, i.e. KILL after 30s
* If a connection can not be established to e.g. vnc or something, die hard. I see error messages like "Connection refused" but nobody cares

I did not observe problems when ending the process with SIGQUIT

**Pull request merged**